infra/images/podman-prod/nginx.nix

```diff
@@ -0,0 +1,88 @@
+{ config, pkgs, ... }:
+
+{
+  users.users.nginx = {
+    extraGroups = [ "acme" ];
+  };
+
+  services.nginx = {
+    enable = true;
+
+    # Use recommended settings
+    recommendedGzipSettings = true;
+    recommendedOptimisation = true;
+    recommendedProxySettings = true;
+    recommendedTlsSettings = true;
+
+    # Only allow PFS-enabled ciphers with AES256
+    sslCiphers = "AES256+EECDH:AES256+EDH:!aNULL";
+
+    commonHttpConfig = ''
+    # Add HSTS header with preloading to HTTPS requests.
+    # Adding this header to HTTP requests is discouraged
+    map $scheme $hsts_header {
+        https   "max-age=31536000; includeSubdomains; preload";
+    }
+    add_header Strict-Transport-Security $hsts_header;
+
+    # Enable CSP for your services.
+    #add_header Content-Security-Policy "script-src 'self'; object-src 'none'; base-uri 'none';" always;
+
+    # Minimize information leaked to other domains
+    add_header 'Referrer-Policy' 'origin-when-cross-origin';
+
+    # Disable embedding as a frame
+    add_header X-Frame-Options DENY;
+
+    # Prevent injection of code in other mime types (XSS Attacks)
+    add_header X-Content-Type-Options nosniff;
+
+    # Enable XSS protection of the browser.
+    # May be unnecessary when CSP is configured properly (see above)
+    add_header X-XSS-Protection "1; mode=block";
+
+    # This might create errors
+    proxy_cookie_path / "/; secure; HttpOnly; SameSite=strict";
+    '';
+
+    
+
+    virtualHosts = let gameNightConfig = {
+    useACMEHost = "gamenight.gay";
+    forceSSL = false;
+    addSSL = true;
+    sslCertificate = "/var/lib/acme/gamenight.gay/cert.pem";
+    locations."/.well-known/acme-challenge/" = {
+        root = "/var/lib/acme/acme-challenge";
+        extraConfig =
+            "if ($scheme = 'https') { rewrite ^ http://$http_host$request_uri? permanent; }";
+    };
+    locations."/" = {
+        proxyPass = "http://127.0.0.1:2727";
+        proxyWebsockets = true; # needed if you need to use WebSocket
+        extraConfig =
+        # required when the target is also TLS server with multiple hosts
+        "proxy_ssl_server_name on;" +
+        # required when the server wants to use HTTP Authentication
+        "proxy_pass_header Authorization;";
+
+        #"if ($scheme = 'http') { rewrite ^ https://$http_host$request_uri? permanent; }";
```

Comment:
Commented out config

Reply:
I can probably just pull this out for now. Was trying to force HTTPS but couldn't get this working yet so that the Let's Encrypt challenge could access the URL over HTTP. There are a few more nix options to maybe play with here for the NGINX config so sorting this out can be a future card for some security hardening that needs done anyway.